parsec/remote_dep_mpi.c

```diff
@@ -2138,39 +2081,52 @@ remote_dep_mpi_get_end_cb(parsec_comm_engine_t *ce,
     return 1;
 }
 
-static int
-remote_dep_ce_init(parsec_context_t* context)
+/**
+ * @brief Called in the context of the communication thread once a change in the
+ * configuration has been noticed. This allows the full reconfiguration of the
+ * communication engine, including the allocation of the necessary structures on
+ * the correct memory node.
+ * 
+ * @param context 
+ * @return int mostly PARSEC_SUCCESS
+ */
+int remote_dep_ce_reconfigure(parsec_context_t* context)
 {
-    int rc;
-
-    if( NULL != parsec_remote_dep_cb_data_mempool ) {
-        /* already fully initialized */
-        return 0;
+    if( NULL != parsec_mpi_same_pos_items ) {
+        free(parsec_mpi_same_pos_items); parsec_mpi_same_pos_items = NULL;
+        parsec_mpi_same_pos_items_size = 0;
     }
-    /* Do this first to give a chance to the communication engine to define
-     * who this process is by setting the corresponding info in the
-     * parsec_context.
-     */
-    if( NULL == parsec_comm_engine_init(context) ) {
-        parsec_warning("Communication engine failed to start. Additional information might be available in the corresponding error message");
-        return PARSEC_ERR_NOT_FOUND;
-    }
-
     /**
      * Finalize the initialization of the upper level structures
      * Worst case: one of the DAGs is going to use up to
      * MAX_PARAM_COUNT times nb_nodes dependencies.
      */
     remote_deps_allocation_init(context->nb_nodes, MAX_PARAM_COUNT);
 
-    PARSEC_OBJ_CONSTRUCT(&dep_activates_fifo, parsec_list_t);
-    PARSEC_OBJ_CONSTRUCT(&dep_activates_noobj_fifo, parsec_list_t);
-    PARSEC_OBJ_CONSTRUCT(&dep_put_fifo, parsec_list_t);
-
     parsec_mpi_same_pos_items_size = context->nb_nodes + (int)DEP_LAST;
+    assert( NULL == parsec_mpi_same_pos_items );
     parsec_mpi_same_pos_items = (dep_cmd_item_t**)calloc(parsec_mpi_same_pos_items_size,
                                                         sizeof(dep_cmd_item_t*));
 
+    if(1 < context->nb_nodes) {
+        /* if nb_nodes==1, the parsec comm engine does not run with its own thread, so don't change the thread
```

Comment:
The comment describes the opposite of what the code does

Reply:
not the opposite, but the other case.